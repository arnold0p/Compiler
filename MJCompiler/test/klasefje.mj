// Test301

program test301
	char niz[];
	const int size = 26;
	int i,j;
	
	int a[];
	
	class Klasa {
		int k;
		int niz[];
		char z;
		{
			int fja(int x,char y){
			 	k = x;
			 	this.k = x + k ;
			 	return k;
			}
		}
	}
	
	class Klasa1 extends Klasa {
		{
			int sum(int a,int b){
				z='a';
				return a + b;
			}
			int fja(int x,char y){
			 	k = x;
			 	this.k = (x + k)*(-2) ;
			 	
			 	
			 	return sum(k,k);
			}
		}
	}
	
	Klasa k;
	
{
	
	void proba(Klasa k){
		k.z = 'X';
	}
	void main()	
	{
	
		
		i=-1;
		//k = new Klasa;
		//i = k.fja(5,'?');
		
		k = new Klasa1;
		i = k.fja(5,'?');
		print(i);
		
		print(k.k);
		
		print(k.z);
		
		proba(k);
		
		print(k.z);
		
	}
}