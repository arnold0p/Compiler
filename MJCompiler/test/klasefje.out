  -40  -20    a    X
Completion took 0 ms